File: backend-api/src/main/java/com/example/usermanagement/entity/UserHealthProfile.java
package com.example.usermanagement.entity;

import javax.persistence.*;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;
import java.time.LocalDateTime;

/**
 * 用户健康档案实体类
 */
@Entity
@Table(name = "user_health_profile")
@EntityListeners(AuditingEntityListener.class)
public class UserHealthProfile {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "profile_id")
    private Long profileId;

    @Column(name = "user_id", unique = true, nullable = false)
    private Long userId;

    @Column(name = "age", nullable = false)
    private Integer age;

    @Column(name = "gender", nullable = false, length = 10)
    private String gender; // male, female, other

    @Column(name = "height", nullable = false)
    private Float height; // 身高(cm)

    @Column(name = "weight", nullable = false)
    private Float weight; // 体重(kg)

    @Column(name = "health_conditions", columnDefinition = "TEXT")
    private String healthConditions; // 健康状况（JSON格式）

    @Column(name = "allergies", length = 500)
    private String allergies; // 过敏史

    @Column(name = "medications", length = 500)
    private String medications; // 当前用药

    @Column(name = "activity_level", length = 20)
    private String activityLevel = "moderate"; // sedentary, light, moderate, active, very_active

    @Column(name = "sugar_limit")
    private Float sugarLimit = 50.0f; // 每日糖分限制(g)

    @Column(name = "calorie_limit")
    private Float calorieLimit = 2000.0f; // 每日热量限制(kcal)

    @Column(name = "water_goal")
    private Float waterGoal = 2000.0f; // 每日饮水目标(ml)

    @CreatedDate
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @LastModifiedDate
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    // 默认构造函数
    public UserHealthProfile() {
    }

    // Getter 和 Setter 方法
    public Long getProfileId() {
        return profileId;
    }

    public void setProfileId(Long profileId) {
        this.profileId = profileId;
    }

    public Long getUserId() {
        return userId;
    }

    public void setUserId(Long userId) {
        this.userId = userId;
    }

    public Integer getAge() {
        return age;
    }

    public void setAge(Integer age) {
        this.age = age;
    }

    public String getGender() {
        return gender;
    }

    public void setGender(String gender) {
        this.gender = gender;
    }

    public Float getHeight() {
        return height;
    }

    public void setHeight(Float height) {
        this.height = height;
    }

    public Float getWeight() {
        return weight;
    }

    public void setWeight(Float weight) {
        this.weight = weight;
    }

    public String getHealthConditions() {
        return healthConditions;
    }

    public void setHealthConditions(String healthConditions) {
        this.healthConditions = healthConditions;
    }

    public String getAllergies() {
        return allergies;
    }

    public void setAllergies(String allergies) {
        this.allergies = allergies;
    }

    public String getMedications() {
        return medications;
    }

    public void setMedications(String medications) {
        this.medications = medications;
    }

    public String getActivityLevel() {
        return activityLevel;
    }

    public void setActivityLevel(String activityLevel) {
        this.activityLevel = activityLevel;
    }

    public Float getSugarLimit() {
        return sugarLimit;
    }

    public void setSugarLimit(Float sugarLimit) {
        this.sugarLimit = sugarLimit;
    }

    public Float getCalorieLimit() {
        return calorieLimit;
    }

    public void setCalorieLimit(Float calorieLimit) {
        this.calorieLimit = calorieLimit;
    }

    public Float getWaterGoal() {
        return waterGoal;
    }

    public void setWaterGoal(Float waterGoal) {
        this.waterGoal = waterGoal;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }

    public LocalDateTime getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(LocalDateTime updatedAt) {
        this.updatedAt = updatedAt;
    }
}

File: app/src/main/java/com/example/myapplication/model/UserHealthProfile.kt
package com.example.myapplication.model

import com.google.gson.annotations.SerializedName

/**
 * 用户健康档案数据模型
 */
data class UserHealthProfile(
    @SerializedName("profile_id")
    val profileId: Long? = null,
    
    @SerializedName("user_id")
    val userId: Long,
    
    @SerializedName("age")
    val age: Int,
    
    @SerializedName("gender")
    val gender: String, // male, female, other
    
    @SerializedName("height")
    val height: Float, // 身高(cm)
    
    @SerializedName("weight")
    val weight: Float, // 体重(kg)
    
    @SerializedName("health_conditions")
    val healthConditions: String? = null, // 健康状况（JSON格式）
    
    @SerializedName("allergies")
    val allergies: String? = null, // 过敏史
    
    @SerializedName("medications")
    val medications: String? = null, // 当前用药
    
    @SerializedName("activity_level")
    val activityLevel: String? = "moderate", // sedentary, light, moderate, active, very_active
    
    @SerializedName("sugar_limit")
    val sugarLimit: Float? = 50f, // 每日糖分限制(g)
    
    @SerializedName("calorie_limit")
    val calorieLimit: Float? = 2000f, // 每日热量限制(kcal)
    
    @SerializedName("water_goal")
    val waterGoal: Float? = 2000f, // 每日饮水目标(ml)
    
    @SerializedName("created_at")
    val createdAt: String? = null,
    
    @SerializedName("updated_at")
    val updatedAt: String? = null
)

/**
 * 健康档案创建/更新请求
 */
data class HealthProfileRequest(
    @SerializedName("age")
    val age: Int,
    
    @SerializedName("gender")
    val gender: String,
    
    @SerializedName("height")
    val height: Float,
    
    @SerializedName("weight")
    val weight: Float,
    
    @SerializedName("health_conditions")
    val healthConditions: String? = null,
    
    @SerializedName("allergies")
    val allergies: String? = null,
    
    @SerializedName("medications")
    val medications: String? = null,
    
    @SerializedName("activity_level")
    val activityLevel: String? = "moderate",
    
    @SerializedName("sugar_limit")
    val sugarLimit: Float? = 50f,
    
    @SerializedName("calorie_limit")
    val calorieLimit: Float? = 2000f,
    
    @SerializedName("water_goal")
    val waterGoal: Float? = 2000f
)

/**
 * 活动水平枚举
 */
enum class ActivityLevel(val value: String, val displayName: String) {
    SEDENTARY("sedentary", "久坐不动"),
    LIGHT("light", "轻度活动"),
    MODERATE("moderate", "中度活动"),
    ACTIVE("active", "活跃"),
    VERY_ACTIVE("very_active", "非常活跃");
    
    companion object {
        fun fromValue(value: String): ActivityLevel {
            return values().find { it.value == value } ?: MODERATE
        }
    }
}


